### DISCUSSION TITLE
How do I cast Bool to UInt32

### DISCUSSION THREAD

**Discussion by tomas16** [mitsuba-renderer/mitsuba3#971] [Q&A] [answered]
I'm a bit embarrassed to ask this question, but I couldn't figure it out.

I have a Float array `x`, and what I want to do is:

```python
y = mi.UInt32(x < 0)
```

`y` can't be a Bool because I'm using it in an indexing expression.

I get this error message:
`drjit.Exception: Refusing to do an extremely inefficient element-by-element array conversion from type <class 'drjit.llvm.ad.Bool'> to <class 'drjit.llvm.ad.UInt'>. Did you forget a cast or detach operation?`

My workaround for now is:

```python
y = dr.select(x < 0, 1, 0)
```

**Comment by njroussel** [comment, marked as answer]
Hi @tomas16 !

Similarly to #576, I don't think there is another solution than using `dr.select` in Python.

**Reply by merlinND** [reply]
An alternative would be `y = mi.UInt32(1) & (x < 0)`, but it's roughly the same and not necesarily more readable.

**Comment by tomas16** [comment]
Thanks! I asked because I thought there would be a cast operation based on that error message.
